### PR TITLE
Adding more tests

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: qp_qiime2/__init__.py [hunk 1]
```diff
@@ -77,6 +80,8 @@
                     req_params[ename] = ('string', etype)
                     # deleting so we don't count it as part of the inputs
                     del inputs[pname]
+                    # we are gonna continue so we don't add this element twice
```
Comment (charles-cowart) [inline qp_qiime2/__init__.py]: gonna = going to

File: qp_qiime2/__init__.py [hunk 2]
```diff
@@ -163,35 +168,59 @@
...
                 if data_type == 'choice' and default is None:
                     qname = q2plugin.name
                     mid = m.id
-                    if qname == 'diversity' and mid == 'alpha_rarefaction':
-                        vals = list(ALPHA_DIVERSITY_METRICS.keys())
+                    error_msg = ("There is an unexpected method (%s %s) with "
+                                 "a choice parameter (%s), without default" % (
+                                    qname, mid, element.description))
+                    # if we are in the diversity choice option, we might want
+                    # to replace the "nerd" names for user friendly ones
```
Comment (charles-cowart) [inline qp_qiime2/__init__.py]: "nerd" = technical?

File: qp_qiime2/__init__.py [hunk 3]
```diff
@@ -163,35 +168,59 @@
...
+                        vals = {
+                            'alpha': am,
+                            'alpha_phylogenetic': amp,
+                            'beta': bm,
+                            'beta_phylogenetic': bmp,
+                            'beta_phylogenetic_alt': bmpa,
+                            'alpha_rarefaction': am.union(amp),
+                            'beta_rarefaction': bm.union(bmp)
+                        }
+                        if mid not in mid:
```
Comment (charles-cowart) [inline qp_qiime2/__init__.py]: This should be changed - it's confusing even if it does actually work.

File: qp_qiime2/tests/test_qiime2.py [hunk 4]
```diff
@@ -224,9 +228,112 @@ def test_beta_phylogenetic(self):
...
+            'diversity should be computed.': '8',
+            'The number of jobs to use for the computation. This works '
+            'by breaking down the pairwise matrix into n_jobs even slices '
+            'and computing them in parallel. If -1 all CPUs are used. If '
+            '1 is given, no parallel computing code is used at all, which '
+            'is useful for debugging. For n_jobs below -1, (n_cpus + 1 + '
+            'n_jobs) are used. Thus for n_jobs = -2, all CPUs but one are '
+            'used. (Description from sklearn.metrics.pairwise_distances)': '1',
+            'qp-hide-method': 'beta',
+            'qp-hide-paramThe beta diversity metric to be computed.': 'metric',
```
Comment (charles-cowart) [inline qp_qiime2/tests/test_qiime2.py:248]: Just to confirm, is this okay, with no spaces or ' etc?
_'qp-hide-paramThe beta diversity..._

**Review by charles-cowart — CHANGES_REQUESTED** [review]

File: qp_qiime2/__init__.py (hunk 3 above)
Comment (antgonza) [reply qp_qiime2/__init__.py]: I actually wrote it like that to make it easier to follow; do you have a suggestion?

File: qp_qiime2/tests/test_qiime2.py (hunk 4 above)
Comment (antgonza) [reply qp_qiime2/tests/test_qiime2.py:248]: yup

File: qp_qiime2/__init__.py (hunk 3 above)
Comment (antgonza) [reply qp_qiime2/__init__.py]: ... or perhaps more explanation?

Comment (charles-cowart) [reply qp_qiime2/__init__.py]: 'if mid not in mid'? At face value, how can it not be in itself?

Comment (charles-cowart) [reply qp_qiime2/__init__.py]: Or how can it be in itself? It reads a little confusing.

Comment (antgonza) [reply qp_qiime2/__init__.py]: duh! didn't see it until now, fixing!

File: .travis.yml [hunk 5]
```diff
@@ -30,10 +30,10 @@ install:
   - source activate qiime2
   - pip install coveralls flake8
   - pip install https://github.com/qiita-spots/qiita_client/archive/master.zip
-  # we are gonna use the same qiime2 environmen to install qtp-diversity
+  # we are going to use the same qiime2 environmen to install qtp-diversity
```
Comment (charles-cowart) [inline .travis.yml]: environmen = environment

File: .travis.yml [hunk 6]
```diff
@@ -30,10 +30,10 @@ install:
...
   - pip install https://github.com/qiita-spots/qtp-diversity/archive/master.zip
   - configure_diversity_types --env-script "source activate qiime2" --server-cert $QIITA_SERVER_CERT
-  # we are gonna use the same qiime2 environmen to install qtp-visualization
+  # we are goint to use the same qiime2 environmen to install qtp-visualization
```
Comment (charles-cowart) [inline .travis.yml]: goint = going

**Review by charles-cowart — CHANGES_REQUESTED** [review]

**Review by charles-cowart — APPROVED** [review]

File: qp_qiime2/qp_qiime2.py [hunk 7]
```diff
@@ -52,15 +52,10 @@
     qiime2.core.type.primitive._MetadataColumnExpression: 'string',
 }
 
-STATE_UNIFRAC_METRICS = {
-    "unweighted UniFrac": "unweighted_unifrac",
-    "weighted normalized UniFrac": "weighted_normalized_unifrac",
-    "weighted unnormalized UniFrac": "weighted_unifrac",
-    "generalized UniFrac": "generalized_unifrac",
+ALPHA_DIVERSITY_METRICS_PHYLOGENETIC = {
+    "Faith's Phylogenetic Diversity": "faith_pd",
```
Comment (charles-cowart) [inline qp_qiime2/qp_qiime2.py:56]: Confirmed, it's legal to have a trailing comma when declaring a dict.

**Review by charles-cowart — APPROVED** [review]

**Review by charles-cowart — APPROVED** [review]

**Pull request merged**